### PR TITLE
Expand GobblinTrackingAuditProcessor to handle metadata audit events.

### PR DESCRIPTION
**Pull request by mars-lan**

### CODE REVIEW

File: wherehows-backend/app/models/kafka/GobblinTrackingAuditProcessor.java [hunk 1]
```diff
@@ -13,23 +13,52 @@
...
+  private static final String DATASET_URN_PREFIX = "hdfs://";
+  private static final String DATASET_OWNER_SOURCE = "IDPC";
+
+  // TODO: Make these regex patterns wh_etl_job_properties
+  private static final Pattern LOCATION_PREFIX_PATTERN = Pattern.compile("/[^/]+(/[^/]+)?");
+
+  private static final List<Pattern> PARENT_PATTERNS = ImmutableList.<Pattern>builder()
+      .add(Pattern.compile("/data/external/gobblin/(.+)"))
+      .add(Pattern.compile("/data/(databases|dbchange|external)/.+"))
+      .add(Pattern.compile("/data/(databases|dbchange|external)/.+"))
```
Comment (alyiwang) [inline wherehows-backend/app/models/kafka/GobblinTrackingAuditProcessor.java]: Duplicate with line 47

File: wherehows-backend/app/models/daos/DatasetDao.java [hunk 2]
```diff
@@ -49,58 +49,57 @@
...
-  public static final String GET_DATASET_DEPENDENTS_IN_OBJ_MAP_TABLE_BY_NAME = "SELECT " +
-          "c.object_dataset_id as dataset_id, d.urn, d.dataset_type, c.object_sub_type " +
-          "FROM cfg_object_name_map c JOIN dict_dataset d ON c.object_dataset_id = d.id " +
-          "WHERE c.mapped_object_type = ? and " +
-          "(c.mapped_object_name = ? or c.mapped_object_name like ?)";
-
-  public static Map<String, Object> getDatasetById(int datasetId)
-    throws SQLException {
+  public static final String GET_DATASET_ID_IN_MAP_TABLE_WITH_TYPE_AND_CLUSTER =
+      "SELECT " + "c.object_dataset_id as dataset_id, d.urn, d.dataset_type, "
```
Comment (alyiwang) [inline wherehows-backend/app/models/daos/DatasetDao.java]: may remove the '+' here

File: wherehows-backend/app/models/daos/DatasetDao.java [hunk 3]
```diff
@@ -49,58 +49,57 @@
...
+  private final static String GET_DATASET_DEPENDS_VIEW =
+      "SELECT object_type, object_sub_type, " + "object_name, map_phrase, is_identical_map, mapped_object_dataset_id, "
+          + "mapped_object_type,  mapped_object_sub_type, mapped_object_name "
+          + "FROM cfg_object_name_map WHERE object_dataset_id = ?";
+
+  public final static String GET_DATASET_URN_PROPERTIES_LIKE_EXPR =
+      "select urn from dict_dataset where properties like :properties";
+
+  public static final String GET_DATASET_DEPENDENTS_IN_OBJ_MAP_TABLE_BY_ID =
+      "SELECT " + "c.object_dataset_id as dataset_id, d.urn, d.dataset_type, c.object_sub_type "
```
Comment (alyiwang) [inline wherehows-backend/app/models/daos/DatasetDao.java]: may remove the '+' here

File: wherehows-backend/app/models/daos/DatasetDao.java [hunk 4]
```diff
@@ -49,58 +49,57 @@
...
+          + "FROM cfg_object_name_map c JOIN dict_dataset d ON c.object_dataset_id = d.id "
+          + "WHERE c.mapped_object_dataset_id = ?";
+
+  public static final String GET_DATASET_DEPENDENTS_IN_OBJ_MAP_TABLE_BY_NAME =
+      "SELECT " + "c.object_dataset_id as dataset_id, d.urn, d.dataset_type, c.object_sub_type "
```
Comment (alyiwang) [inline wherehows-backend/app/models/daos/DatasetDao.java]: may remove the '+' here

File: wherehows-backend/app/models/daos/DatasetDao.java [hunk 5]
```diff
@@ -49,58 +49,57 @@
...
+          + "and lower(c.object_type) = ? and lower(i.server_cluster) = ?";
+
+  public static final String GET_DATASET_ID_IN_MAP_TABLE_WITH_CLUSTER = "SELECT c.object_dataset_id as dataset_id, "
+      + "d.urn, d.dataset_type, i.deployment_tier, i.data_center, i.server_cluster "
+      + "FROM cfg_object_name_map c JOIN dict_dataset d ON c.object_dataset_id = d.id "
+      + "LEFT JOIN dict_dataset_instance i ON c.object_dataset_id = i.dataset_id "
+      + "WHERE c.object_dataset_id is not null and  lower(c.object_name) = ? and lower(i.server_cluster) = ?";
+
+  private final static String GET_DATASET_DEPENDS_VIEW =
+      "SELECT object_type, object_sub_type, " + "object_name, map_phrase, is_identical_map, mapped_object_dataset_id, "
```
Comment (alyiwang) [inline wherehows-backend/app/models/daos/DatasetDao.java]: may remove the '+' here

File: wherehows-backend/app/models/daos/DatasetDao.java [hunk 6]
```diff
@@ -49,58 +49,57 @@
...
+          + "mapped_object_type,  mapped_object_sub_type, mapped_object_name "
```
Comment (alyiwang) [inline wherehows-backend/app/models/daos/DatasetDao.java]: might remove the double space before 'mapped_object_sub_type' as well

File: wherehows-backend/app/models/kafka/GobblinTrackingAuditProcessor.java (hunk 1 above)
Comment (mars-lan) [inline wherehows-backend/app/models/kafka/GobblinTrackingAuditProcessor.java]: Thanks for catching this.

File: wherehows-backend/app/models/daos/DatasetDao.java (hunk 2 above)
Comment (mars-lan) [inline wherehows-backend/app/models/daos/DatasetDao.java]: Done

File: wherehows-backend/app/models/daos/DatasetDao.java (hunk 5 above)
Comment (mars-lan) [inline wherehows-backend/app/models/daos/DatasetDao.java]: Done

File: wherehows-backend/app/models/daos/DatasetDao.java (hunk 3 above)
Comment (mars-lan) [inline wherehows-backend/app/models/daos/DatasetDao.java]: Done

File: wherehows-backend/app/models/daos/DatasetDao.java (hunk 4 above)
Comment (mars-lan) [inline wherehows-backend/app/models/daos/DatasetDao.java]: Done

File: wherehows-backend/app/models/daos/DatasetDao.java (hunk 6 above)
Comment (mars-lan) [inline wherehows-backend/app/models/daos/DatasetDao.java]: Done

**Pull request merged**